### PR TITLE
M2P-537 Add support for Magento native In-Store Pickup step 2

### PR DESCRIPTION
**Pull request by dumega**
# Description

As we decide to add support BOPIS Spilt shipping to M2, and In-store delivery method is supported with version 2.4.0 of Magento (https://devdocs.magento.com/guides/v2.4/inventory/release-notes.html#v120) , this PR is for adding support for this feature.

Step 2 : Add in-store pickup options to split shipping endpoint if they are available.

Fixes: https://boltpay.atlassian.net/browse/M2P-537

#changelog Add support for Magento native In-Store Pickup step 2

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [x] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [ ] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# For PR Reviewer 
- [ ] Reviewed unit tests to make sure we are using real components rather than mocks as much as possible?
- [ ] For any major change (observer, new Bolt feature, core Magento interaction) we must add a feature switch, did you verify this?

# Checklist:

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have created or modified unit tests to sufficiently cover my changes.
- [ ] I have added my Jira ticket link and provided a changelog message.

### CODE REVIEW

**Review by danac-gs — APPROVED** [review]

File: Model/Api/Shipping.php [hunk 1]
```diff
@@ -86,17 +86,21 @@ public function __construct(
     /**
      * @param array $addressData
      * @param null $shipping_option
+     * @param null $ship_to_store_option
      * @return ShippingDataInterface
      * @throws LocalizedException
      */
-    public function generateResult($addressData, $shipping_option)
+    public function generateResult($addressData, $shipping_option, $ship_to_store_option)
```
Comment (vitaliyreznikov) [inline Model/Api/Shipping.php:93]: Looks like we do not use new parameter `$ship_to_store_option` as well as existing parameter `$shipping_option`

Comment (vitaliyreznikov) [inline Model/Api/Shipping.php:93]: Could we add comment about why we don't use the second and the third parameters?

**Review by vitaliyreznikov — APPROVED** [review]

File: Model/Api/Shipping.php (hunk 1 above)
Comment (dumega) [reply Model/Api/Shipping.php:93]: @vitaliyreznikov we have an abstract function in its parent class ShippingTax, both shipping api class and tax api class extend this abstract class, 
```
abstract public function generateResult($addressData, $shipping_option, $ship_to_store_option); 
```
And for shipping api hook, it only sends addressData in the request, then the tax api hook has shipping_option or ship_to_store_option data.

Comment (dumega) [reply Model/Api/Shipping.php:93]: Added

**Pull request merged**